src/Client.ts

```diff
@@ -161,7 +175,7 @@ export class Client {
                 .do((msg: IMessage) => {
                     if (msg instanceof OpenMessage) {
                         this.currentRetryCount = 0;
-                        o.roles = o.roles || Client.roles();
```

Comment:
These roles should not be overridable by the user since they are announcing the client's capabilities to the router and the user can not change the capabilities.

Reply:
Yeah, that makes sense. With these changes the "bugs" would be fixed and the library would be perfect for using in my project again, thank you.

Let me know if there's anything you want me to do before merging this pull request.